src/combined-selectors/reporting-config-search-results.ts

```diff
@@ -41,16 +67,37 @@ function searchReportingConfig(
 			feature.keywords?.some( ( keyword ) => includesIgnoringCase( keyword, searchTerm ) );
 
 		if ( includesIgnoringCase( feature.name, searchTerm ) || keywordsIncludeSearchTerm() ) {
-			searchResults.features.add( featureId );
-
-			const feature = features[ featureId ];
-			if ( feature.parentType === 'product' ) {
-				searchResults.products.add( feature.parentId );
-			} else {
-				searchResults.featureGroups.add( feature.parentId );
-				const parentFeatureGroup = featureGroups[ feature.parentId ];
-				searchResults.products.add( parentFeatureGroup.productId );
+			addFeatureAndParents( featureId );
+		}
+	}
+
+	// Finally, search for the search term in the description of the products and feature areas
+	const searchTermTokens = tokenizeAndNormalize( searchTerm );
+	const scores: Record< string, { type: TaskParentEntityType; score: number } > = {};
+
+	for ( const token of searchTermTokens ) {
+		const matchingEntities = invertedIndex[ token ] || [];
+
+		for ( const { type, id, weight } of matchingEntities ) {
+			if ( ! scores[ id ] ) {
+				scores[ id ] = { type, score: 0 };
 			}
+			scores[ id ].score += weight;
+		}
+	}
+
+	const scoreThreshold = 1;
+	const addToSearchResultsByType = {
+		product: ( entityId: string ) => searchResults.products.add( entityId ),
+		featureGroup: ( entityId: string ) => addFeatureGroupAndParents( entityId ),
+		feature: ( entityId: string ) => addFeatureAndParents( entityId ),
+	};
+
+	for ( const entityId in scores ) {
+		if ( scores[ entityId ].score >= scoreThreshold ) {
+			const entityType = scores[ entityId ].type;
+			const addToSearchResults = addToSearchResultsByType[ entityType ];
+			addToSearchResults( entityId );
```

Comment:
**Praise** Oh this is just a beautiful block of code! 😍  I really like how you grouped the functions to add the entities together here!